### PR TITLE
Feature/add def upload

### PR DESCRIPTION
**Pull request by gbhojraj**

### CODE REVIEW

File: src/Courses/CourseCreation/CourseEditPage.tsx [hunk 1]
```diff
@@ -1,41 +1,73 @@
...
+            let course = await AxiosRequest.get(`/curriculum/${courseId}`);
+
+            console.log(course.data.data);
+            setCourse(course.data.data);
+        })();
+    }, []);
+
+
+    const callShowTopicCreation = (unit: number, e: any = null) => {
+        if (e != null) {
```
Comment (tommylettieri) [inline src/Courses/CourseCreation/CourseEditPage.tsx:36]: why single equals?

Comment (tommylettieri) [reply src/Courses/CourseCreation/CourseEditPage.tsx:36]: null and undefined I would guess, but does this also catch things like false?  Even though it should be fine I feel like we should keep up with good practice, doesn't linter yell about this?

File: src/Courses/CourseDetailsPage.tsx [hunk 2]
```diff
@@ -18,14 +21,62 @@ enum CourseDetailsTabs {
...
         <Container>
-            <Tabs defaultActiveKey={CourseDetailsTabs.TOPICS} id="course-details-tabs">
+            <Tabs defaultActiveKey={CourseDetailsTabs.DETAILS} id="course-details-tabs">
+                <Tab eventKey={CourseDetailsTabs.DETAILS} title="Details">
+                    {course && (
+                        <>
+                            <h1>{course.name}</h1>
+                            <p>Course description content goes here.</p>
+                            <Button onClick={() => copyToClipboard()}>Copy Student Enrollment Link</Button>
+                            <a ref={a => enrollLinkRef = a} id='enrollUrlLink' href={enrollUrl}>https://{enrollUrl}</a>
```
Comment (tommylettieri) [inline src/Courses/CourseDetailsPage.tsx:70]: hardcoding https?

File: src/Courses/CourseDetailsTabs/TopicsTab.tsx [hunk 3]
```diff
@@ -1,32 +1,29 @@
...
-    {name: 'Unit 3', topics: mock_topics, unit_id: 3},
-];
-
 export const TopicsTab: React.FC<TopicsTabProps> = () => {
+    const { courseId } = useParams();
+    const [course, setCourse] = useState<any>(null);
+
+    useEffect(() => {
+        (async ()=>{
+            let course = await AxiosRequest.get(`/curriculum/${courseId}`);
```
Comment (tommylettieri) [inline src/Courses/CourseDetailsTabs/TopicsTab.tsx:17]: Is there any reason we didn't separate server integration into it's own files

Comment (gbhojraj) [reply src/Courses/CourseDetailsTabs/TopicsTab.tsx:17]: What do you mean, `server integration`? Are you asking why we don't pull out all the API requests into a separate file? We could have created named functions to pull in and pass into useEffect, but I'm not sure there's an easy way to handle state with that.

File: src/Courses/CourseCreation/CourseEditPage.tsx (hunk 1 above)
Comment (gbhojraj) [reply src/Courses/CourseCreation/CourseEditPage.tsx:36]: Using soft equals is fine where it's intentional. In this case, I want any falsy value to fail. e should be an event, anyhow.

File: src/Courses/CourseDetailsPage.tsx (hunk 2 above)
Comment (gbhojraj) [reply src/Courses/CourseDetailsPage.tsx:70]: Is this a problem?

**Pull request merged**